### PR TITLE
Feat/validate deps exist

### PR DESCRIPTION
**Pull request by ekbsnyk**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Handles edge case where user tests a Pipfile project with no dependencies, or with only dev-deps but no --dev-deps flag; throws informative error instead of just exiting silently without informing user why.
Next I'll be replicating this in pip-deps.

#### What are the relevant tickets?
Addresses https://snyk.zendesk.com/agent/tickets/6872

#### Screenshots
Screenshot 2020-10-27 at 16 54 38

### CODE REVIEW

File: test/workspaces/pipfile-dev-deps-only/Pipfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
```
Comment (AlbertKogan) [inline test/workspaces/pipfile-dev-deps-only/Pipfile:6]: Customer's problem occurred when `Pipfile` did not contain `packages` at all. Worth to test this scenario as well

File: pysrc/pip_resolve.py [hunk 2]
```diff
@@ -199,9 +199,21 @@ def get_requirements_list(requirements_file_path, dev_deps=False):
...
+        req_list = list(parsed_reqs.get('packages') or [])
         if dev_deps:
-            req_list.extend(parsed_reqs.get('dev-packages', []))
+            req_list.extend(parsed_reqs.get('dev-packages') or [])
+        
+        # Error informatively if there are no dependencies
+        if req_list == []:
+            if parsed_reqs['dev-packages'].__len__() > 0 and dev_deps == False:
+                msg = 'No requirements found. Note: dev-deps present but skipped as --dev flag is false.'
+                print(msg)
```
Comment (AlbertKogan) [inline pysrc/pip_resolve.py:210]: Seems like this should be removed🧐

File: pysrc/pip_resolve.py [hunk 3]
```diff
@@ -199,9 +199,21 @@ def get_requirements_list(requirements_file_path, dev_deps=False):
...
             requirements_data = f.read()
         parsed_reqs = pipfile.parse(requirements_data)
-        req_list = list(parsed_reqs.get('packages', []))
+        req_list = list(parsed_reqs.get('packages') or [])
         if dev_deps:
-            req_list.extend(parsed_reqs.get('dev-packages', []))
+            req_list.extend(parsed_reqs.get('dev-packages') or [])
+        
+        # Error informatively if there are no dependencies
+        if req_list == []:
```
Comment (AlbertKogan) [inline pysrc/pip_resolve.py:207]: According to `PEP8` better to use `not` here: `if not req_list:`

File: pysrc/pip_resolve.py [hunk 4]
```diff
@@ -199,9 +199,21 @@ def get_requirements_list(requirements_file_path, dev_deps=False):
...
+            if parsed_reqs['dev-packages'].__len__() > 0 and dev_deps == False:
```
Comment (AlbertKogan) [inline pysrc/pip_resolve.py:208]: `parsed_reqs['dev-packages'].__len__()` might throw `AttributeError: 'NoneType' object has no attribute '__len__'` in case if `dev_packages` is `None`
Other way to do this if statement might be like this: `if parsed_reqs.get('dev-packages') and not dev_deps:`

File: pysrc/pip_resolve.py (hunk 3 above)
Comment (ekbsnyk) [reply pysrc/pip_resolve.py:207]: 👍 Didn't realise an empty list evaluates to False.

File: pysrc/pip_resolve.py (hunk 2 above)
Comment (ekbsnyk) [reply pysrc/pip_resolve.py:210]: Good catch!

File: pysrc/pip_resolve.py (hunk 4 above)
Comment (ekbsnyk) [reply pysrc/pip_resolve.py:208]: Looks much more idiomatic too. Thanks :)
